drivers/ethernet/Kconfig.stm32_hal

```diff
@@ -49,34 +50,38 @@ config ETH_STM32_HAL_PHY_ADDRESS
 	  The phy address to use.
 
 config ETH_STM32_HAL_RANDOM_MAC
-	bool "Random MAC address"
+	bool "Random MAC address (DEPRECATED)"
 	depends on ENTROPY_GENERATOR
-	default y
+	select DEPRECATED
 	help
 	  Generate a random MAC address dynamically.
+	  Deprecated in favor of device tree property "zephyr,random-mac-address"
 
 if !ETH_STM32_HAL_RANDOM_MAC
 
 config ETH_STM32_HAL_MAC3
-	hex "MAC Address Byte 3"
+	hex "MAC Address Byte 3 (DEPRECATED)"
 	default 0
 	range 0 0xff
 	help
 	  This is the byte 3 of the MAC address.
+	  Deprecated in favor of device tree property "local-mac-address"
 
```

Comment:
The problem with this is there this file shows 2 possible options, the deprecated random mac, or the 3 byte mac address, it does not allow for a third option. The 3 bytes should be in an if() that only shows if that mode is selected, the 3 modes are random (deprecated), static (deprecated) and dts - so the first option here can be made a choice with those 3 options where dts is the default

Reply:
Problem is that current status (before changes) is:
- Default config: Static address used (using default 0)
- If selected: random defined

Adding a choice, user would have to select the new symbol 'static kconfig' to be able to use the deprecated `ETH_STM32_HAL_MACX` symbols. So modify his configuration to be able use a deprecated configuration....
To me the point of deprecation, is to let user keeping his configuration w/o doing nothing, but be informed this wll have to be changed.
If he's forced to change his configuration anyway, no point of using the deprecated symbols..

Reply:
@nordicjm Ok, I tried to follow your advise. Please have a new look.